### PR TITLE
feat: OSemgrep Supply Chain

### PR DESCRIPTION
**Pull request by mmcqd**

### CODE REVIEW

File: libs/ast_generic/AST_generic.ml [hunk 1]
```diff
@@ -2427,3 +2427,15 @@ class virtual ['self] map_legacy =
       let skind = self#visit_stmt_kind env x.s in
       { x with s = skind }
   end
+
+type dependency = {
```
Comment (aryx) [inline libs/ast_generic/AST_generic.ml:2449]: probably better out of libs/, in src/core/Supply_chain.ml or src/core/ProjectDependency.ml

Comment (aryx) [reply libs/ast_generic/AST_generic.ml:2449]: or PackageDependency.ml maybe given the field names.

File: src/core/Xtarget.ml [hunk 2]
```diff
@@ -13,4 +13,14 @@ type t = {
   lazy_content : string lazy_t;
   (* This is valid only for xlang = Xlang.L ..., not for LRegex|LGeneric *)
   lazy_ast_and_errors : (AST_generic.program * Tok.location list) lazy_t;
+  lockfile_data : lockfile_data option;
+}
+
+and lockfile_data = {
```
Comment (aryx) [inline src/core/Xtarget.ml:19]: Maybe we should use a variant then? or not abuse Xtarget.ml and have a separate type?
src/core/Lockfile.ml ?

**Review by aryx — COMMENTED** [review]
Exciting!

File: src/core/Rule.ml [hunk 3]
```diff
@@ -463,6 +463,21 @@ and step = {
...
 
+(*****************************************************************************)
+(* Supply Chain *)
+(*****************************************************************************)
+
+(* You can only do single layer deep OR *)
+type dependency_formula = dependency_pattern list
+
+and dependency_pattern = {
+  (* TODO: actually good types *)
```
Comment (aryx) [inline src/core/Rule.ml:585]: You can reuse the types in semgrep_output_v1.atd.
It's now accessible from src/core, so you can do

ecosystem: Out.ecosystem

(Out is an alias defined at the top of Rule.ml)

**Comment by mmcqd** [comment]
closed in favor of https://github.com/semgrep/semgrep/pull/9695

**Pull request closed without merging**